person.js

```diff
@@ -80,6 +86,40 @@ var Person = new Lang.Class({
         }));
     },
 
+    _getGitLabInfo: function() {
+        let _httpSession = new Soup.Session({user_agent: 'jwendell/gnome-shell-extension-timezone'});
+        let url = 'https://gitlab.com/api/v4/avatar?email=%s'.format(this.gitlab);
+        let message = new Soup.Message({method: 'GET', uri: new Soup.URI(url)});
+        if (this._gitlabToken) {
+            message.request_headers.append("Authorization", "token " + this._gitlabToken);
+        }
+
+        _httpSession.queue_message(message, Lang.bind(this, function(session, message) {
+            if (message.status_code != Soup.KnownStatusCode.OK) {
+                log('Response code "%d" getting data from gitlab for user email address %s. Got: %s'.format(message.status_code, this.github, message.response_body.data));
+                return;
+            }
+            let p;
+            try {
+                p = JSON.parse(message.response_body.data);
+            } catch (e) {
+                log('Error parsing gitlab email address response for user %s: %s'.format(this.gitlab, e));
+                return;
+            }
+
+            if (!this.avatar && p.avatar_url)
+                this.avatar = p.avatar_url;
+
+            if (!this.name && p.name)
```

Comment:
Looking at https://docs.gitlab.com/ee/api/avatar.html it seems the `avatar_url` is the only field in the response.
Perhaps you could use the `/users` endpoint and get all fields like we do with github?